### PR TITLE
feat(serverless-contracts): use authorizerType to enforce authorizer in lambda config

### PR DESCRIPTION
**Pull request by fargito**
Fixes: #224

### CODE REVIEW

File: user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md [hunk 1]
```diff
@@ -104,11 +105,11 @@ In the `config.ts` file of our lambda, in the `events` section, we need to use t
 export default {
   environment: {},
   handler: getHandlerPath(__dirname),
-  events: [getTrigger(myContract)],
+  events: [getTrigger(myContract, {})],
```
Comment (adriencaccia) [inline user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md:108]: Is there no way to have the type inference without having to add the parameter ?
We can maybe ask @ThomasAribart, the TS 🧙

Comment (fargito) [reply user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md:108]: I don't know unfortunately 🤔

Comment (fargito) [reply user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md]: @adriencaccia the TS 🧙🏻 has helped me and the argument is no longer required for unauthenticated contracts! Many thanks @ThomasAribart 💪🏻

File: user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md [hunk 2]
```diff
@@ -132,6 +133,34 @@ export default {
 };
 ```
 
+The static typing also enforces authenticated contracts by requiring authorizers.
+
+For an unauthenticated contract (`authorizerType: undefined`):
+
+```ts
+export default {
```
Comment (adriencaccia) [inline user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md:141]: We should add the contract in the code example, to make it clearer

File: user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md [hunk 3]
```diff
@@ -132,6 +133,34 @@ export default {
...
+      authorizer: 'arn:aws:...', // typescript will throw an error
+    }),
+  ],
+};
+```
+
+For an authenticated contract:
+
+```ts
+export default {
```
Comment (adriencaccia) [inline user-docs/documentation/docs/how-to-guides/1-use-serverless-contracts/1-api-gateway.md:155]: Same comment

**Review by adriencaccia — CHANGES_REQUESTED** [review]

**Review by adriencaccia — APPROVED** [review]

**Pull request merged**